frontend/ui/widget/qrmessage.lua

```diff
@@ -0,0 +1,117 @@
+--[[--
+Widget that displays a qr code.
+
+It vanishes on key press or after a given timeout.
+
+Example:
+    local UIManager = require("ui/uimanager")
+    local _ = require("gettext")
+    local Screen = require("device").screen
+    local sample
+    sample = QRMessage:new{
+        text = _("my message"),
+        height = Screen:scaleBySize(400),
+        width = Screen:scaleBySize(400),
+        timeout = 5,  -- This widget will vanish in 5 seconds.
+    }
+    UIManager:show(sample)
+]]
+
+local Blitbuffer = require("ffi/blitbuffer")
+local CenterContainer = require("ui/widget/container/centercontainer")
+local Device = require("device")
+local FrameContainer = require("ui/widget/container/framecontainer")
+local Geom = require("ui/geometry")
+local GestureRange = require("ui/gesturerange")
+local QRWidget = require("ui/widget/qrwidget")
+local InputContainer = require("ui/widget/container/inputcontainer")
+local UIManager = require("ui/uimanager")
+local Input = Device.input
+local Screen = Device.screen
+
+local QRMessage = InputContainer:new{
+    modal = true,
+    timeout = nil, -- in seconds
+    text = nil,  -- The text to encode.
+    width = nil,  -- The width. Keep it nil to use original width.
```

Comment:
if width or height are nil the thingy crash

```
./luajit: frontend/ui/widget/qrmessage.lua:65: attempt to perform arithmetic on field 'width' (a nil value)
stack traceback:
	frontend/ui/widget/qrmessage.lua:65: in function 'init'
	frontend/ui/widget/widget.lua:48: in function 'new'
	frontend/ui/elements/common_settings_menu_table.lua:19: in function 'callback'
	frontend/ui/widget/touchmenu.lua:802: in function 'action'
	frontend/ui/uimanager.lua:876: in function '_checkTasks'
	frontend/ui/uimanager.lua:1213: in function 'handleInput'
	frontend/ui/uimanager.lua:1306: in function 'run'
	./reader.lua:300: in main chunk
	[C]: at 0x55f4a029979e
```

Reply:
Should be fixed.